Docker - with Ubuntu 16.04

dpshelio opened the pull request:
This adds a docker container :whale2: for the old version running on 16.04 (now the same than Travis!).

Here we've also updated the readme and Travis to be homogenous with the docker. The installation instructions have been brought inside the repository to keep it in sync with what's currently there (and not pointed to the wiki as that becomes misleading with different versions).

dpshelio committed:
Removed flags that are unrecognized options on all dune used

dpshelio committed:
Add information on installation about the attempted version tested

dpshelio committed:
:pencil2: `std` not being used

dpshelio committed:
Adds Docker that builds and executes the tests as expected

dpshelio committed:
Renames INSTALL and gets it consistent around the rest of the files

dpshelio committed:
Adds installation instructions within the repository - easier to track

dpshelio committed:
Update instructions to include docker

dpshelio committed:
Sort travis file to keep in sync with the container instructions

dpshelio committed:
Update to Zoltan 3.83 the docker and other missing bits

dpshelio committed:
Cleaned up the installation instructions to allign them to the Docker and Travis

dpshelio committed:
Keeping all options for PETSc similar

dpshelio committed:
Update Travis configuration to use xenial and the preferred options

dpshelio committed:
[ci] Unset the CC variables as they are known to interfere with MPI

https://docs.travis-ci.com/user/languages/cpp/#mpi-projects

dpshelio committed:
:whale2: :pencil2: - fix zoltan file to download

dpshelio committed:
[ci] Test to make PETSc without making it all

dpshelio committed:
[ci] sudify all the installation for petsc and zoltan

dpshelio committed:
[ci] delete and append lines with sed, need of two different commands

giordano commented on README.md:
````diff
@@ -1,7 +1,51 @@
+
+Then you can run the container as:
+```bash
+docker run -it --name mypeits -v /full/localpath:/mydata  uclrits/peits:16.04
````
As a Docker-illiterate, it's not clear to me what `/full/localpath:/mydata` should be and if I need to replace with something else (David privately told me what `/mydata` is and that the user shouldn't change it).  Just pasting the command the shell seems to work.

giordano reviewed (commented):
I could build and successfully use the docker container by following the instructions in the `README.md`.  I left a minor comment about some paths mentioned in that file, though.

I wonder whether it makes sense to move to or duplicate in `README.md` the test installation instructions that are in `INSTALL.md`.  Another option is point the user to those instructions from the `README.md`.  They're useful, but they're not immediately discoverable from the `README.md` if the user used the Docker container.

dpshelio committed:
[docker][ci][doc] Updated PETSC repository location

dpshelio committed:
[ci] Try to build a cache and use it

It also "cleans" the travis script by externalising some of the installation
steps as scripts under `.ci`.

dpshelio committed:
[ci] Removes one nn_test run to create cache

dpshelio committed:
[ci] Moved cache to a user writable

So the only need for sudo is to run `make install`

dpshelio committed:
[ci] trying to get MUMPS building on cache

When building petsc it may download the dependencies on a place
that's not being cached properly. Hopefully, running make will
download them again

dpshelio committed:
[ci] installs zoltan and petsc on a cached directory

dpshelio committed:
[ci] only build dependencies when creating the cache

dpshelio committed:
Revert "[ci] Removes one nn_test run to create cache"

This reverts commit ab46202ae4ca5ec2183d3091f9e41287e81d7939.

dpshelio committed:
Rename config files with `cfg` extension and propagate these changes to other files.

dpshelio reviewed (commented):
@Jimbles this is an "important" change. I've simply done it so we get colours when opening the configuration files. Could you check whether I'm not missing anything?

This refers to that last commit!

dpshelio committed:
Adds information about running PEITS and fixes some problems

Thanks @giordano for your review

dpshelio committed:
[docker] Makes it smaller in size

Removes the source and building folders for peits and zoltan

dpshelio committed:
[doc] update to the right docker tag that is going to be once we relased

dpshelio committed:
[ci] removes printing logs as they aren't needed

dpshelio committed:
[ci] Removes one nn_test run - double run doesn't fail

giordano committed:
Make tests executable scripts

giordano committed:
Make tests executable scripts (#22)

Make tests executable scripts

giordano committed:
Start bash at login instead of zsh

dpshelio committed:
Start bash at login instead of zsh (#23)

Start bash at login instead of zsh
